### PR TITLE
Replace all simple nqp::if(42,foo,bar) with ternaries

### PR DESCRIPTION
**Pull request by lizmat**
If memory serves me well, about 1.5 years ago, optimizations in
Rakudo made the construct:

   42 ?? foo !! bar

equally fast as the nqp equivalent:

  nqp::if(42,foo,bar)

This commit converts all of the simple nqp::if() structures in the
core.c setting to their ternary equivalents.

- binary size of CORE.c.setting.moarvm file is unchanged (almost)
- speed should be on par with the nqp::if() equivalents
- readability of the core setting should be improved.

make test and make spectest clean.

### CODE REVIEW

**Review by AlexDaniel — APPROVED** [review]
Last time I checked there was still a consistent difference, even if minor. Maybe that's no longer true. Either way I find the readability improvement much more important. Looked through the changes and didn't find any accidental mistakes. Great work!

File: src/core.c/BagHash.pm6 [hunk 1]
```diff
@@ -149,7 +145,7 @@ my class BagHash does Baggy {
                 nqp::if(
                   nqp::existskey(elems,$key),
                   nqp::getattr(nqp::atkey(elems,$key),Pair,'$!value'),
-                  0
+                  # 0 the value of existskey if the key doesn't exist
```
Comment (AlexDaniel) [inline src/core.c/BagHash.pm6:148]: Ah btw I don't know if this change was intended to be here.

Comment (lizmat) [reply src/core.c/BagHash.pm6:148]: It is...   to make it similar to the approach for Bag.AT-KEY

Comment (AlexDaniel) [reply src/core.c/BagHash.pm6:148]: OK! Just looked different in an otherwise repetitive diff, so made me wonder.

**Pull request merged**